### ISSUE TITLE
Adding New Nav Item dynamically

### ISSUE THREAD

**Issue by jefferyleo** [BilalShahid13/PersistentBottomNavBar#9]
I've encountered the `RangeError (index): Invalid value: Not in range 0...3, inclusive 4` after updated the version to 1.4.4
Generally I'm using Flutter Provider to notify the bottom navigation bar to detect once the user has login, I'll be added a new item into the bottom navigation bar, it works perfectly in 1.2.1 previously. However since I updated to 1.4.4 version, this method seems like not working anymore, 

The link below is a simple demo on how I added the bottom nav bar item dynamically by using this plugin and Provider.
https://drive.google.com/open?id=1dtL0q_tbCmpnlEjyq7CjSnjqL4EuYaOs
Press the login button, a new nav item will appear, click it and will show the error below

The screenshot is here....(which is on another project)
![WhatsApp Image 2020-04-29 at 19 43 23](https://user-images.githubusercontent.com/6831609/80592327-d93f2680-8a51-11ea-8f2e-01aa5085573b.jpeg)
The last item is added dynamically by notifying the Provider

**Things I suspect...**

I've notify my Provider values to update the item into my bottom navigation bar, but it doesn't update the navigation item count to the original class itself, so this happened.

**Things I tried...**
Flutter Clean with uninstalling the app from the device.
I've revert back the version to 1.2.1 and try out my app again, the error is still persist.

I hope you could provide me a better solution on how to update the bottom navigation bar item dynamically. Hope my information is given enough. Thanks a lot and appreciated.

Just in case you need the Flutter Doctor
```
[flutter] flutter doctor -v
[√] Flutter (Channel master, v1.16.3, on Microsoft Windows [Version 10.0.17763.1098], locale en-US)
    • Flutter version 1.16.3 at C:\Users\JefferyLeo\Documents\Flutter\flutter
    • Framework revision be3a4b37b3 (5 weeks ago), 2020-03-26 18:08:05 -0700
    • Engine revision b2bdeb3f0f
    • Dart version 2.8.0 (build 2.8.0-dev.17.0 1402e8e1a4)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
    • Android SDK at C:\Users\JefferyLeo\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 43.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] VS Code (version 1.44.2)
    • VS Code at C:\Users\JefferyLeo\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.9.1

[√] Connected device (1 available)
    • SM N950F • ce0817180d63385d0c7e • android-arm64 • Android 8.0.0 (API 26)

• No issues found!
exit code 0
```

**Comment by BilalShahid13** [comment]
This is probably due to the change I made during catering custom widgets. I'll test your code and get back to you as soon as I can.

Thanks for reporting the issue.

**Comment by jefferyleo** [comment]
@BilalShahid13 Okay thanks a lot. Wait for your good news! I appreciated that this plugin is really nice and useful, will keep supporting you

**Comment by BilalShahid13** [comment]
The issue is fixed! Please check the version `1.5.1`. Also your support is much appreciated 👍 

This thread is now being closed.
